fix: Admin/Owner not able to see attendees info in team seated events

asadath1395 opened the pull request:
## What does this PR do?

Fixes admins/Owners not able to see attendees info in team seated events

- Fixes #22802 (GitHub issue number)
- Fixes CAL-6172 (Linear issue number - should be visible at the bottom of the GitHub issue description)

## Updates since last revision

This PR was stale and has been updated to resolve merge conflicts with upstream/main:

**Merge conflicts resolved in:**
- `apps/web/modules/bookings/views/bookings-single-view.getServerSideProps.tsx`
- `apps/web/playwright/booking-seats.e2e.ts`
- `packages/trpc/server/routers/viewer/bookings/get.handler.ts`

**Import path updates:**
- Replaced `isTeamAdmin` with `isTeamOwner` (original function no longer exists in upstream)
- Updated `isOrganisationAdmin` import path to `@calcom/features/pbac/utils/isOrganisationAdmin`

**E2E test improvements:**
- Added `data-testid` attribute to Attendee component in `BookingListItem.tsx` using email for unique identification
- Updated attendee visibility tests in `bookings-list.e2e.ts` to use `page.getByTestId()` instead of brittle text locators

## Visual Demo (For contributors especially)

#### Image Demo (if applicable):

**Before**

**Owner**
<img width="1426" height="861" alt="image" src="https://github.com/user-attachments/assets/b406f921-f581-4736-92ac-78dca933214b" />

**Admin**
<img width="1426" height="861" alt="image" src="https://github.com/user-attachments/assets/67920c91-1010-4e9b-ab89-e79495b65210" />


**After**

**Owner**
<img width="1426" height="861" alt="Screenshot 2025-08-20 at 12 37 22 PM" src="https://github.com/user-attachments/assets/9e74895d-705d-419f-a07c-f249283692e8" />

**Admin**
<img width="1426" height="861" alt="image" src="https://github.com/user-attachments/assets/f4dd0a4c-b172-45bc-a460-03faca4900bf" />


## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

1. Create a team with seated event type where `seatsShowAttendees` is set to `false`
2. Have multiple attendees book the same time slot
3. Verify that team owners can see all attendees
4. Verify that team admins can see all attendees
5. Verify that organization admins can see all attendees
6. Verify that regular team members (non-hosts) only see themselves

## Human Review Checklist

- [ ] **Verify team admin visibility**: The change from `isTeamAdmin` to `isTeamOwner` may have semantic implications. `isTeamOwner` only checks for OWNER role. Confirm that team admins (not just owners) can still see attendees - the org admin check via `isOrganisationAdmin` does check both ADMIN and OWNER roles.
- [ ] Run the E2E tests in `booking-seats.e2e.ts` and `bookings-list.e2e.ts` to verify the new tests pass
- [ ] Verify the new `data-testid` attribute renders correctly with email addresses containing special characters (e.g., `first+seats@cal.com`)

---
Link to Devin run: https://app.devin.ai/sessions/0b79cdec258d4890be8b62957864c28b
Requested by: unknown ()
<!-- devin-review-badge-begin -->

---

<a href="https://app.devin.ai/review/calcom/cal.com/pull/23205" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    <img src="https://static.devin.ai/assets/gh-open-in-devin-review-light.svg?v=1" alt="Open with Devin">
  </picture>
</a>
<!-- devin-review-badge-end -->

asadath1395 committed:
fix: Admin/Owner not able to see attendees info in team seated events

asadath1395 commented:
@anikdhabal Could you please review this PR since you raised the issue. Thanks!

kart1ka committed:
Merge branch 'main' into fix/admin-owner-attendees

kart1ka reviewed (dismissed):
Left a few comments. Please also add tests for this fix.

asadath1395 committed:
Merge branch 'main' of https://github.com/calcom/cal.com into fix/admin-owner-attendees

asadath1395 commented on apps/web/modules/bookings/views/bookings-single-view.getServerSideProps.tsx:
```diff
@@ -157,26 +159,41 @@ export async function getServerSideProps(context: GetServerSidePropsContext) {
+      return true;
+    }
+
+    if (eventType.teamId) {
+      const hasTeamOrOrgPermissions = await checkTeamOrOrgPermissions(
+        userId,
+        eventType.teamId,
+        eventType.team?.parent?.id
+      );
+      if (hasTeamOrOrgPermissions) {
+        return true;
+      }
+    }
```
This PR only deals with team seated events, hence the check.

asadath1395 committed:
Address review comments

asadath1395 commented:
@kart1ka I have addressed your comments. Please review it whenever you are available. Thanks! Appreciate all the work you are doing on my PRs.

asadath1395 committed:
Merge branch 'main' into fix/admin-owner-attendees

asadath1395 committed:
Merge branch 'main' into fix/admin-owner-attendees

anikdhabal commented on apps/web/playwright/lib/test-helpers/teamHelpers.ts:
```diff
@@ -48,3 +50,91 @@ export async function createRoundRobinTeamEventType({
 }
+
+/**
+ * Sets up a team and booking seats for testing team role visibility
+ * @param user - The original user who created the event
+ * @param booking - The booking to associate with the team
+ * @param teamUser - The user to assign the specified team role
+ * @param role - The team role to assign (ADMIN, MEMBER, or OWNER)
+ * @returns Object containing the created team and booking seats
+ */
```
Remove these comments

anikdhabal committed:
Merge branch 'main' into fix/admin-owner-attendees

anikdhabal committed:
made few changes

anikdhabal committed:
update

anikdhabal committed:
Update get.handler.ts

anikdhabal committed:
remove comments

anikdhabal reviewed (dismissed):
Looks good

anikdhabal reviewed (dismissed):
LGTM

anikdhabal committed:
addresed coderrabit commnet

volnei commented on apps/web/playwright/lib/test-helpers/teamHelpers.ts:
```diff
@@ -48,3 +50,83 @@ export async function createRoundRobinTeamEventType({
+  teamUser: { id: number },
+  role: "ADMIN" | "MEMBER" | "OWNER"
+) {
+  const bookingWithEventType = await prisma.booking.findFirst({
```
uid is unique, so findUnique is more coherent/semantic.

volnei reviewed (dismissed):
Nit comment but good to fix :)

asadath1395 committed:
Merge branch 'main' of https://github.com/calcom/cal.com into fix/admin-owner-attendees

asadath1395 committed:
Address review comments

asadath1395 commented:
@anikdhabal Appreciate making changes to the PR. Not sure why you removed a centralised function `checkIfUserIsHost` and copied the logic over in multiple places, isn't DRY better?

anikdhabal committed:
Merge branch 'main' into fix/admin-owner-attendees

asadath1395 committed:
Merge branch 'main' of https://github.com/calcom/cal.com into fix/admin-owner-attendees

asadath1395 committed:
Merge branch 'main' of https://github.com/calcom/cal.com into fix/admin-owner-attendees

kart1ka committed:
Merge branch 'main' into fix/admin-owner-attendees

kart1ka committed:
fix: org admins able to see seat info

kart1ka committed:
fix

anikdhabal reviewed (dismissed):
Looks good

asadath1395 committed:
Merge branch 'main' into fix/admin-owner-attendees

kart1ka committed:
fix

kart1ka committed:
Merge branch 'main' into fix/admin-owner-attendees

pallava-joshi reviewed (commented):
can you please resolve the merge conflicts.

Pallava-Joshi commented:
@cubic-dev-ai leave a review

devin-ai-integration[bot] committed:
Merge upstream/main into fix/admin-owner-attendees

Resolved merge conflicts in:
- apps/web/modules/bookings/views/bookings-single-view.getServerSideProps.tsx
- apps/web/playwright/booking-seats.e2e.ts
- packages/trpc/server/routers/viewer/bookings/get.handler.ts

Combined imports and preserved both the PR's admin/owner attendee visibility
fix and upstream's team member visibility features.

Co-Authored-By: unknown <>

devin-ai-integration[bot] committed:
fix: Update imports to use isTeamOwner and correct isOrganisationAdmin path

- Replace isTeamAdmin with isTeamOwner (isTeamAdmin no longer exists in upstream)
- Update isOrganisationAdmin import path to @calcom/features/pbac/utils/isOrganisationAdmin
- Remove deprecated @calcom/lib/server/queries/organisations import

Co-Authored-By: unknown <>

keithwillcode committed:
Update apps/web/playwright/bookings-list.e2e.ts

Co-authored-by: cubic-dev-ai[bot] <191113872+cubic-dev-ai[bot]@users.noreply.github.com>

devin-ai-integration[bot] committed:
fix: Replace brittle text locators with data-testid in E2E tests

- Add data-testid attribute to Attendee component using email for unique identification
- Update attendee visibility tests to use page.getByTestId() instead of text locators
- Follows Playwright best practice of using resilient data-testid selectors

Co-Authored-By: unknown <>

sahitya-chandra reviewed (commented):
@asadath1395 Are you still working on this?
making this draft until then
